s3_storage/models/s3storage.py

```diff
@@ -50,64 +63,52 @@ def _get_storage_client(self):
-        except ResponseError as e:
-            _logger.info("_read_file (s3) reading %s", fname, exc_info=True)
-        return r
+                res = response.read().encode('base64')
+        except ResponseError as errn:
+            LOGGER.info("_read_file (s3) reading %s", fname, exc_info=True)
+        return res
 
     @api.model
     def _file_write(self, value):
+        if self._storage() != 's3':
+            return super(S3Attachment, self)._file_write(value)
         client, bucket = self._get_storage_client()
         bin_value = value.decode('base64')
         fname, _ = self._get_path(bin_value)
         try:
             client.put_object(
                 bucket, fname, StringIO(bin_value), len(bin_value))
-        except ResponseError as e:
-            _logger.info("_file_write (s3) writing %s", fname, exc_info=True)
+        except ResponseError as errn:
+            LOGGER.info("_file_write (s3) writing %s", fname, exc_info=True)
         return fname
 
     @api.model
-    def _file_gc(self):
-        if self._storage() != 'file':
-            return
+    def _file_delete(self, fname):
+        if self._storage() != 's3':
+            return super(S3Attachment, self)._file_delete()
         client, bucket = self._get_storage_client()
 
-        # See comment in the original implementation
-        cr = self._cr
-        cr.commit()
-        self._cr.execute("LOCK ir_attachment IN SHARE MODE")
-
-        self._cr.execute("SELECT fname FROM s3_garbage")
-        fnames = [r[0] for r in cr.fetchall()]
-        kept = set()
+        self.env.cr.execute("""
+        SELECT COUNT(*) FROM ir_attachment WHERE store_fname = %s""", (fname,))
+        fnames = [r[0] for r in self.env.cr.fetchall()]
         try:
             errors = client.remove_objects(bucket, fnames)
             for error in errors:
-                _logger.info("_file_gc (s3) removing %s: %s - %s",
-                             error.object_name,
-                             error.error_code,
-                             error.error_message)
-                kept.add(error.object_name)
-            # mark as removed
-            self._cr.execute(
-                "DELETE FROM s3_garbage WHERE fname NOT IN %s",
-                (tuple(kept),))
-        except ResponseError as e:
-            _logger.info("_file_gc (s3)", exc_info=True)
-        cr.commit()
-        _logger.info("filestore gc %d checked, %d removed",
-                     len(fnames), len(fnames)-len(kept))
- 
-    @api.model
-    def _mark_for_gc(self, fname):
-        self._cr.execute("INSERT INTO s3_garbage VALUES (%s) "
-                   "ON CONFLICT DO NOTHING")
+                LOGGER.info("_file_gc (s3) removing %s: %s - %s",
+                            error.object_name,
+                            error.error_code,
+                            error.error_message)
+        except ResponseError as errn:
+            LOGGER.info("_file_gc (s3)", exc_info=True)
```

Comment:
Should be changed to `_file_delete`.

Reply:
Changed on #6c28db81de75a3b191a0eaabdafc246cdec7a1c0

Reply:
fixed on 6c28db8